src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OperatorIndexerCompletionProviderTests.cs

```diff
@@ -0,0 +1,500 @@
+namespace N1.Nested1
+{{
+    public class C
+    {{
+    }}
+}}
+
+namespace N1.Nested2
+{{
+    public class C
+    {{
+    }}
+}}
+namespace N2
+{{
+    public class Conversion
+    {{
+        public static explicit operator N1.Nested1.C(Conversion _) => new N1.Nested1.C();
+        public static explicit operator N1.Nested2.C(Conversion _) => new N1.Nested2.C();
+    }}
+}}
+namespace N1
+{{
+    {usingDirective}
+    public class Test
+    {{
+        public void M()
+        {{
+            var conversion = new N2.Conversion();
+            conversion.$$
+        }}
+    }}
+}}
+", SourceCodeKind.Regular);
+            Assert.Collection(items,
+                i => Assert.Equal(displayText1, i.DisplayText),
+                i => Assert.Equal(displayText2, i.DisplayText));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+
+        public async Task ExplicitUserDefinedConversionIsSuggestedForAllExplicitConversionsToOtherTypesAndNotForImplicitConversions()
+        {
+            var items = await GetCompletionItemsAsync(@"
+public class C
+{
+    public static explicit operator float(C c) => 0;
+    public static explicit operator int(C c) => 0;
+    
+    public static explicit operator C(float f) => new C();
+    public static implicit operator C(string s) => new C();
+    public static implicit operator string(C c) => "";
+}
+
+public class Program
+{
+    public void Main()
+    {
+        var c = new C();
+        c.$$
+    }
+}
+", SourceCodeKind.Regular);
+            Assert.Collection(items,
+                i => Assert.Equal("(float)", i.DisplayText),
+                i => Assert.Equal("(int)", i.DisplayText));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+
+        public async Task ExplicitUserDefinedConversionFromOtherTypeToTargetIsNotSuggested()
+        {
+            await VerifyNoItemsExistAsync(@"
+public class C
+{
+    public static explicit operator C(float f) => new C();
+}
+
+public class Program
+{
+    public void Main()
+    {
+        float f = 1;
+        f.$$
+    }
+}
+");
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task ExplicitUserDefinedConversionIsApplied()
+        {
+            await VerifyCustomCommitProviderAsync(@"
+public class C
+{
+    public static explicit operator float(C c) => 0;
+}
+
+public class Program
+{
+    public void Main()
+    {
+        var c = new C();
+        c.$$
+    }
+}
+", "(float)", @"
+public class C
+{
+    public static explicit operator float(C c) => 0;
+}
+
+public class Program
+{
+    public void Main()
+    {
+        var c = new C();
+        ((float)c).$$
+    }
+}
+");
+        }
```

Comment:
isn't thsi the same as previous tests?

Reply:
> isn't thsi the same as previous tests?

No. This tests the application of a suggestion. The previous test, on the other hand, shows, that if

* class C defines an explicit conversion from float to C (the other way around)
* the conversion is **not** suggested if the user types `somefloat.$$`

This is because we can not find the conversion in `float`. Instead, we need to look into every other type in the semanticModel to find that conversion. I don't think we should do this.